virttest/iscsi.py

```diff
@@ -834,5 +836,497 @@ def create_iSCSI(params, root_dir=data_dir.get_tmp_dir()):
+                if bool(re.findall("%s$" % self.target, iscsi_discover(ip), re.M)):
+                    iscsi_login(self.target, ip)
+
+
+class MultiPathsIscsiLIOManager(object):
+    """
+    Manager for the MultiPathsIscsiLIO.
+    """
+
+    def __init__(self, params, root_dir):
+        """
+        Init the object based on the params and root_dir.
+
+        :param params: parameters of iSCSI
+        :type params: dict
+        :param root_dir: path for image
+        :type root_dir: string
+        """
+        self._targets_mgr_created = []
+        targets = params.objects("iscsi_targets")
+        self._iscsi_targets = {}
+        for target in targets:
+            val = params.object_params(target)["iscsi_target_options"]
+            val = ast.literal_eval(val)
+            self._iscsi_targets[val["name"]] = val
+        bss = params.objects("iscsi_target_backstores")
+        self._iscsi_target_bss = {}
+        for bs in bss:
+            val = params.object_params(bs)["iscsi_target_backstore_options"]
+            self._iscsi_target_bss[bs] = ast.literal_eval(val)
+
+        # stores the objects based on each backstores
+        self._bss_object = {}
+        for name, cfg in self._iscsi_target_bss.items():
+            cfg["backstore_name"] = name
+            self._bss_object[name] = backstore_type[cfg["backend"]](cfg, root_dir)
+
+        # stores the objects based on each targets
+        self._targets_object = {}
+        for name, cfg in self._iscsi_targets.items():
+            single_target_param = copy.deepcopy(cfg)
+            single_target_param["target"] = name
+            single_target_param["backstores"] = {}
+            for bs_name in self._iscsi_targets[name]["binding_basckstores"]:
+                single_target_param["backstores"].update(
+                    {bs_name: self._bss_object[bs_name]}
+                )
+            self._targets_object[name] = MultiPathsIscsiLIO(single_target_param)
+
+    def export_target(self, targets=None):
+        """
+        Create backstores and targets.
+
+        :param targets: the targets.
+        :type targets: list
+        """
+        targets = targets if targets else self._targets_object.keys()
+        for target in targets:
+            for name in self._iscsi_targets[target]["binding_basckstores"]:
+                self._bss_object[name].create_backstore()
+            self._targets_object[target].export_target()
+            self._targets_mgr_created.append(target)
+
+    def query_targets(self, emulated_image=None):
+        """
+        Filter the targets from image name given.
+
+        :param emulated_image: the image name. If none, do NOT check emulated_image
+        :type emulated_image: string
+
+        :return: the targets in list or []
+        :rtype: list
+        """
+        if not emulated_image:
+            return self._targets_mgr_created
+
+        filtered_targets = []
+        for emu in emulated_image:
```

Comment:
it looks emulated_image is a list, would you please update line1192? And if it's a list, what about using emulated_images or emulated_image_list